### ISSUE TITLE
libsystemd cannot be loaded from the statically built binaries on Linux

### ISSUE THREAD

**Issue by mhf-ir** [timberio/vector#1433]
i use amd64 bin file with following config:

```toml

data_dir = "/tmp/vector"

[sources.ss11]
  # REQUIRED
  type = "journald" # example, must be: "journald"
  # units = []
  # current_boot_only = true # default

[sinks.out]
    inputs = ["ss11"]
    type = "console"
    encoding = "json"
```

```bash
$ sudo ./vector test stdio.toml 
Running stdio.toml tests
```

```bash
$ sudo ./vector --config stdio.toml 
Dec 26 14:11:18.218  INFO vector: Log level "info" is enabled.
Dec 26 14:11:18.218  INFO vector: Loading config. path="stdio.toml"
Dec 26 14:11:18.226  INFO vector: Vector is starting. version="0.6.0" git_version="v0.6.0" released="Thu, 12 Dec 2019 17:24:03 +0000" arch="x86_64"
Dec 26 14:11:18.226 ERROR vector::topology: Configuration error: Source "ss11": journald error: dlopen Error: Could not open library: Dynamic loading not supported
```

**Comment by binarylogic** [comment]
Thanks @mhf-ir , we'll see what's goiong on. @a-rodin let me know if @bruceg is better suited to resolve this.

**Comment by mhf-ir** [comment]
I have to test vector replacement for fluenbit for our infrastructure i see your're rush get to issue for cover it up.
Thanks guys again.

**Comment by a-rodin** [comment]
It looks like `dlopen` doesn't work when called from statically linked with `musl` binaries. See https://www.openwall.com/lists/musl/2012/12/08/4 for details.

While we are trying to solve issue before the next version, I suggest compiling Vector from source for `x86_64-unknown-linux-gnu` target, for which the C library would be linked dynamically and `dlopen` would work. See [the build instructions](https://vector.dev/docs/setup/installation/manual/from-source/) for details.

**Comment by mhf-ir** [comment]
Build from source solve my problem but i think this but is problem in most used linux OS on the world:
https://w3techs.com/technologies/details/os-linux

Any update on this?

**Comment by bruceg** [comment]
This is being discussed in issue #1473 and I have just published a proposed fix in PR #1526

**Comment by binarylogic** [comment]
This is resolved with #1526
